src/pin.ts

```diff
@@ -0,0 +1,436 @@
+      }
+    }
+
+    // Create the new version directory structure
+    await this.createVersionDirectory(contractName, version)
+
+    // Copy contract files to the new version directory
+    await this.copyContractFiles(contractPath, contractName, version)
+
+    // Update chelonia.json configuration
+    await this.updateCheloniaConfig(contractName, version, contractPath)
+
+    // Generate manifests for this specific contract version
+    await this.generateContractManifest(contractName, version)
+
+    console.log(colors.green(`✅ Successfully pinned ${contractName} to version ${version}`))
+    console.log(colors.gray(`Location: contracts/${contractName}/${version}/`))
+  }
+
+  /**
+   * Load chelonia.json configuration file
+   * Creates a default one if it doesn't exist
+   */
+  private async loadCheloniaConfig () {
+    const configPath = join(this.projectRoot, 'chelonia.json')
+
+    if (existsSync(configPath)) {
+      try {
+        const configContent = await readFile(configPath, 'utf8')
+        this.cheloniaConfig = JSON.parse(configContent)
+        console.log(colors.blue('📄 Loaded existing chelonia.json'))
+      } catch (error) {
+        console.warn(colors.yellow(`Warning: Could not parse chelonia.json: ${error}`))
+        // Use default empty config
+      }
+    } else {
+      console.log(colors.blue('📄 Creating new chelonia.json'))
+      // Will be saved when updateCheloniaConfig is called
+    }
+
+    // Ensure contracts object exists
+    if (!this.cheloniaConfig.contracts) {
+      this.cheloniaConfig.contracts = {}
+    }
+  }
+
+  /**
+   * Extract contract name from file path
+   * Examples:
+   * - 'frontend/model/contracts/chatroom.js' -> 'chatroom'
+   * - 'contracts/identity.js' -> 'identity'
+   * - 'src/contracts/group.js' -> 'group'
+   */
+  private extractContractName (contractPath: string): string {
+    const fileName = basename(contractPath)
+    // Remove .js extension and any -slim suffix
+    return fileName.replace(/(-slim)?\.js$/, '')
+  }
+
+  /**
+   * Show available contracts in the project
+   * This helps users know what contracts they can pin
+   */
+  private async showAvailableContracts () {
+    console.log(colors.cyan('📋 Available contracts to pin:'))
+    console.log(colors.gray('Usage: chel pin <version> <contract-file-path>'))
+    console.log()
+
+    // Look for contract files in common locations
+    const commonPaths = [
+      'frontend/model/contracts',
+      'contracts',
+      'src/contracts'
+    ]
+
+    let foundContracts = false
+
+    for (const searchPath of commonPaths) {
+      const fullPath = join(this.projectRoot, searchPath)
+      if (existsSync(fullPath)) {
+        try {
+          const files = await readdir(fullPath)
+          const contractFiles = files.filter(f => f.endsWith('.js') && !f.endsWith('-slim.js'))
```

Comment:
My other comment about extensions applies here too. You're supposed to have the manifest already, not create a new one.